src/measure_theory/interval_integral.lean

```diff
@@ -0,0 +1,405 @@
+  { have H1 := H.1, have H2 := H.2,
+    split_ifs,
+    { simp only [interval_of_le h] at *,
+      rw [abs_of_nonneg (integral_on_nonneg_of_ae _), abs_of_nonneg (integral_on_nonneg_of_ae _)],
+      { apply integral_on_le_integral_on_ae, assumption' },
+      { filter_upwards [hf, hfg] λ a hf hfg h, le_trans (hf h) (hfg h) },
+      { filter_upwards [hf] λ a hf, hf } },
+    { simp only [interval_of_le' (le_of_not_ge h)] at *,
+      rw [abs_of_nonpos, abs_of_nonpos, neg_neg, neg_neg],
+      { apply integral_on_le_integral_on_ae, assumption' },
+      { rw [neg_nonpos], apply integral_on_nonneg_of_ae,
+        filter_upwards [hf, hfg] λ a hf hfg h, le_trans (hf h) (hfg h)  },
+      { rw [neg_nonpos], apply integral_on_nonneg_of_ae,
+        filter_upwards [hf] λ a hf, hf } } },
+  rw [integral_undef H, abs_zero],
+  exact abs_nonneg _
+end
+
+lemma abs_integral_le_of_le {f g : ℝ → ℝ}
+  (hf : ∀ x ∈ a..b, 0 ≤ f x) (hfg : ∀ x ∈ a..b, f x ≤ g x)
+  (hgm : measurable_on a..b g) (hgi : integrable_on a..b g) :
+  abs (∫ x in a...b, f x) ≤ abs (∫ x in a...b, g x) :=
+abs_integral_le_of_ae_le (univ_mem_sets' hf) (univ_mem_sets' hfg) hgm hgi
+
+lemma integral_Icc_combine (hab : a ≤ b) (hbc : b ≤ c)
+  (hfm : measurable_on a..b f) (hfi : integrable_on a..b f)
+  (hfm' : measurable_on b..c f) (hfi' : integrable_on b..c f) :
+  (∫ x in Icc a b, f x) + (∫ x in Icc b c, f x) = ∫ x in Icc a c, f x :=
+begin
+  simp only [interval_of_le hab, interval_of_le hbc] at *,
+  have : Icc a c = Icc a b ∪ Icc b c, { rw Icc_union_Icc_eq_Icc; assumption },
+  rw [this, integral_on_union_ae],
+  assumption',
+  repeat { exact is_measurable_Icc },
+  rw all_ae_not_mem_iff,
+  rw Icc_inter_Icc_eq_singleton hab hbc,
+  exact volume_singleton
+end
+
+lemma integral_combine
+  (hfm : measurable_on a..b f) (hfi : integrable_on a..b f)
+  (hfm' : measurable_on b..c f) (hfi' : integrable_on b..c f) :
+  (∫ x in a...b, f x) + (∫ x in b...c, f x) = ∫ x in a...c, f x :=
+begin
+  wlog ac : a ≤ c := le_total a c using [a c, c a],
+  { have acm : measurable_on a..c f := hfm.interval_combine hfm',
+    have aci : integrable_on a..c f := hfi.interval_combine hfm hfm' hfi',
+    split_ifs with ab bc bc,
+    { rw [interval_of_le ab, interval_of_le bc, interval_of_le ac],
+      exact integral_Icc_combine ab bc hfm hfi hfm' hfi' },
+    { rw [add_neg_eq_iff_eq_add, eq_comm],
+      have cb : c ≤ b := le_of_not_ge bc,
+      rw [interval_of_le ab, interval_of_le' cb, interval_of_le ac],
+      refine integral_Icc_combine ac cb acm aci _ _,
+      rwa [interval_swap], rwa [interval_swap] },
+    { rw [neg_add_eq_iff_eq_add, eq_comm],
+      have ba : b ≤ a := le_of_not_ge ab,
+      rw [interval_of_le' ba, interval_of_le ac, interval_of_le bc],
+      refine integral_Icc_combine ba ac _ _ acm aci,
+      rwa [interval_swap], rwa [interval_swap] },
+    { have := le_trans ac (le_of_not_ge bc), contradiction }, },
+    show (∫ x in a...b, f x) + (∫ x in b...c, f x) = ∫ x in a...c, f x,
```

Comment:
The rest of the proof should be either indented with 2 spaces or should be inside `{ ... }`.